libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialogBot.cs

```diff
@@ -0,0 +1,103 @@
+            IConfiguration configuration,
+            ILogger<AdaptiveDialogBot> logger,
+            ResourceExplorer resourceExplorer,
+            IStorage storage = null,
+            BotFrameworkClient botFrameworkClient = null)
+        {
+            _defaultLocale = configuration.GetSection(DefaultLocaleKey).Value ?? DefaultLocale;
+            _rootDialogId = configuration.GetSection(RootDialogKey).Value ?? "Main.dialog";
+
+            _logger = logger;
+            _resourceExplorer = resourceExplorer;
+            _storage = storage ?? new MemoryStorage();
+            _botFrameworkClient = botFrameworkClient;
+        }
+
+        async Task IBot.OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"IBot.OnTurnContext for dialog '{_rootDialogId}'");
+
+            // Set up the TurnState the Dialog is expecting
+            SetUpTurnState(turnContext);
+
+            // Load the Dialog from the ResourceExplorer
+            var rootDialog = await CreateDialogAsync(cancellationToken).ConfigureAwait(false);
+
+            // Run the Dialog
+            await rootDialog.RunAsync(turnContext, turnContext.TurnState.Get<ConversationState>().CreateProperty<DialogState>("DialogState"), cancellationToken).ConfigureAwait(false);
+
+            // Save any updates that have been made
+            await turnContext.TurnState.Get<ConversationState>().SaveChangesAsync(turnContext, false, cancellationToken).ConfigureAwait(false);
+            await turnContext.TurnState.Get<UserState>().SaveChangesAsync(turnContext, false, cancellationToken).ConfigureAwait(false);
+        }
+
+        private void SetUpTurnState(ITurnContext turnContext)
+        {
+            turnContext.TurnState.Add(_botFrameworkClient);
+            turnContext.TurnState.Add<SkillConversationIdFactoryBase>(new SkillConversationIdFactory(_storage));
+            turnContext.TurnState.Add(new ConversationState(_storage));
+            turnContext.TurnState.Add(new UserState(_storage));
+            turnContext.TurnState.Add(_resourceExplorer);
+            turnContext.TurnState.Add<LanguageGenerator>(_resourceExplorer.TryGetResource(DefaultLg, out var resource) ? new ResourceMultiLanguageGenerator(DefaultLg) : new TemplateEngineLanguageGenerator());
+            turnContext.TurnState.Add(_languageGeneratorManagers.GetOrAdd(_resourceExplorer, _ => new LanguageGeneratorManager(_resourceExplorer)));
+            turnContext.TurnState.Add(new LanguagePolicy(_defaultLocale));
+        }
+
+        private async Task<Dialog> CreateDialogAsync(CancellationToken cancellationToken)
+        {
+            if (!_resourceExplorer.TryGetResource(_rootDialogId, out var rootDialogResource))
+            {
+                var msg = $"The ResourceExplorer could not find a resourced with id '{_rootDialogId}'";
+                _logger.LogError(msg);
+                throw new Exception(msg);
```

Comment:
Do not throw generic exceptions (see https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2201), we don't have this FxCop rule enabled but we will probably enable it at some point in the future.
InvalidOperationException is the closest to a generic exception that FxCop won't complain about